### PR TITLE
[fix] metadata issue, fixes #74

### LINKED ISSUE #74: Function calls are not supported in decorators but 'IntercomModule' was called. - Regression in beta.6
_Originally posted by @chrishawn in https://github.com/CaliStyle/ng-intercom/issue_comments#issuecomment-430313444_

### PR DESCRIPTION
**Pull request by lucagouty**
**Summary of changes:** 
- Change imports to fix metadata issue

Honestly, I'm not sure what was the issue, but I found that fix in `ngx-progressbar` which had the same issue some time ago : 
- Issue : https://github.com/MurhafSousli/ngx-progressbar/issues/160
- PR : https://github.com/MurhafSousli/ngx-progressbar/pull/168/files

**Other unrelated changes (I'll rollback if you want)**
- Update npm-run-all (I would keep that one, see https://github.com/dominictarr/event-stream/issues/116 ⚠️)
  _Also, I couldn't make `npm i` to work without it, because `npm ERR! 404 Not Found: flatmap-stream@0.1.1`_
- Add baseUrl to tsconfig (required for "paths")
image

**Intended/example use case:** 

Of course, first I made sure I was reproducing the issue, without this change, I get 
```
ERROR in Error during template compile of 'CoreModule'
  Function calls are not supported in decorators but 'IntercomModule' was called.
: Unexpected value 'undefined' imported by the module 'CoreModule in /some/path/src/app/core/core.module.ts'
Error during template compile of 'CoreModule'
  Function calls are not supported in decorators but 'IntercomModule' was called.
```

With this fix, everything goes well now (Angular7, both `ng build --prod` and `ng xi18n`)

**Checklist:**
- [x] `npm run build` runs without error
- [x] `ng serve` spawns app, Intercom messenger is visible and interactive, and there are no errors in the console

**Closes issue:** #74

### CODE REVIEW

File: public_api.ts [hunk 1]
```diff
@@ -1 +1,9 @@
-export * from './src/app/ng-intercom'
+export { Intercom } from './src/app/ng-intercom/intercom/intercom'
+export { IntercomConfig } from './src/app/ng-intercom/shared/intercom-config'
+export { IntercomHideDirective } from './src/app/ng-intercom/directives/hide.directive'
+export { IntercomModule } from './src/app/ng-intercom/intercom.module'
+export { IntercomShowDirective } from './src/app/ng-intercom/directives/show.directive'
+export { IntercomShowMessagesDirective } from './src/app/ng-intercom/directives/show-messages.directive'
+export { IntercomShowNewMessageDirective } from './src/app/ng-intercom/directives/show-new-message.directive'
+export { IntercomShutdownDirective } from './src/app/ng-intercom/directives/shutdown.directive'
+export { IntercomTrackEventDirective } from './src/app/ng-intercom/directives/track-event.directive'
```
Comment (lucagouty) [inline public_api.ts:9]: I think this should be enough, but I don't want to take a risk, so let me know if you want me to update it : 
```ts
export { Intercom } from './src/app/ng-intercom/intercom/intercom'
export { IntercomConfig } from './src/app/ng-intercom/shared/intercom-config'
// export { IntercomHideDirective } from './src/app/ng-intercom/directives/hide.directive'
export { IntercomModule } from './src/app/ng-intercom/intercom.module'
// export { IntercomShowDirective } from './src/app/ng-intercom/directives/show.directive'
// export { IntercomShowMessagesDirective } from './src/app/ng-intercom/directives/show-messages.directive'
// export { IntercomShowNewMessageDirective } from './src/app/ng-intercom/directives/show-new-message.directive'
// export { IntercomShutdownDirective } from './src/app/ng-intercom/directives/shutdown.directive'
// export { IntercomTrackEventDirective } from './src/app/ng-intercom/directives/track-event.directive'
```

`npm run test` and `ng serve` both work with only those 3 exports (as does our app)

**Comment by scott-wyatt** [comment]
Thank you for the PR! I'm going to close and reopen this to trigger the CI.

**Comment by scott-wyatt** [comment]
This will be released in #84 release 7.0.0-beta.1

**Comment by wbhob** [comment]
thanks so much for your work, Luca !

**Pull request merged**